corehq/motech/openmrs/openmrs_config.py

```diff
@@ -177,6 +177,13 @@ class ObservationMapping(DocumentSchema):
 
 class OpenmrsFormConfig(DocumentSchema):
     xmlns = StringProperty()
+
+    # Used to determine the start of a visit and an encounter. The end
+    # of a visit is set to one day (specifically 23:59:59) later. If not
+    # given, the value defaults to when the form was completed according
+    # to the device, /meta/timeEnd.
```

Comment:
Whats the reason behind taking timeEnd and not timeStart/timeBegin , whichever is added on the form?

Reply:
Good question! I don't actually remember whether I originally wrote this, or @dannyroberts . If it was me, I might rationalize that a mobile worker might start a form before an encounter, and complete it as soon as they were finished with all the questions. I don't know if one can generalise across all use cases, and one could just as easily argue that they start a form at the start of an encounter, and complete it later. I also tend not to trust the clock on a device; I'd rather go with the submission time according to the server.

I could be easily persuaded to change this if you have any points in favour of a different approach.

Reply:
Can't argue with that, thanks :)